### DISCUSSION TITLE
Allows using different elements as scroll containers

### DISCUSSION THREAD

**Discussion by piecyk** [TanStack/virtual#249] [Ideas]
Current it's possible via custom useScroll hook, would be nice to allow passing a ref to element.

**Comment by tannerlinsley** [comment]
We can handle this in the next major version, as I think it would be breaking.

**Reply by uffou** [reply]
This would be great, especially if the container can support multiple lists

**Reply by piecyk** [reply]
Current it's possible via `onScrollElement` there you can pass ref to element on which scroll event will be registered. Checkout the code and PR #210 there is example for window scroller that will work with current main.
